common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -507,43 +492,30 @@ def ensure_user_information(
     # invariants have been violated and future misbehavior is likely.
     user_inactive = user and not user.is_active
     user_unset = user is None
-    dispatch_to_login = is_login and (user_unset or user_inactive)
+
+    dispatch_to_login = (
+        ((is_login or is_login_2) and (user_unset or user_inactive))
+        or
+        ((is_register or is_register_2) and user_inactive)
```

Comment:
Never mind, I see that I was wrong about `user_inactive`: it will be set to False in that case.  I think this looks fine then.